manifests/install.pp

```diff
@@ -13,56 +13,57 @@
-      ensure_packages(['unzip'])
-    }
-    staging::file { 'consul.zip':
-      source => $consul::real_download_url
-    } ->
-    staging::extract { 'consul.zip':
-      target  => $consul::bin_dir,
-      creates => "${consul::bin_dir}/consul",
-    } ->
-    file { "${consul::bin_dir}/consul":
-      owner => 'root',
-      group => 0, # 0 instead of root because OS X uses "wheel".
-      mode  => '0555',
-    }
-
-    if ($consul::ui_dir and $consul::data_dir) {
-      file { "${consul::data_dir}/${consul::version}_web_ui":
-        ensure => 'directory',
-        owner  => 'root',
-        group  => 0, # 0 instead of root because OS X uses "wheel".
-        mode   => '0755',
-      } ->
-      staging::deploy { 'consul_web_ui.zip':
-        source  => $consul::real_ui_download_url,
-        target  => "${consul::data_dir}/${consul::version}_web_ui",
-        creates => "${consul::data_dir}/${consul::version}_web_ui/dist",
+  case $consul::install_method {
+    'url': {
+      if $::operatingsystem != 'darwin' {
+        ensure_packages(['unzip'])
       }
-      file { $consul::ui_dir:
-        ensure => 'symlink',
-        target => "${consul::data_dir}/${consul::version}_web_ui/dist",
+      staging::file { 'consul.zip':
+        source => $consul::real_download_url
+      } ->
+      staging::extract { 'consul.zip':
+        target  => $consul::bin_dir,
+        creates => "${consul::bin_dir}/consul",
+      } ->
+      file { "${consul::bin_dir}/consul":
+        owner => 'root',
+        group => 0, # 0 instead of root because OS X uses "wheel".
+        mode  => '0555',
       }
-    }
-
-  } elsif $consul::install_method == 'package' {
 
-    package { $consul::package_name:
-      ensure => $consul::package_ensure,
+      if ($consul::ui_dir and $consul::data_dir) {
+        file { "${consul::data_dir}/${consul::version}_web_ui":
+          ensure => 'directory',
+          owner  => 'root',
+          group  => 0, # 0 instead of root because OS X uses "wheel".
+          mode   => '0755',
+        } ->
+        staging::deploy { 'consul_web_ui.zip':
+          source  => $consul::real_ui_download_url,
+          target  => "${consul::data_dir}/${consul::version}_web_ui",
+          creates => "${consul::data_dir}/${consul::version}_web_ui/dist",
+        }
+        file { $consul::ui_dir:
+          ensure => 'symlink',
+          target => "${consul::data_dir}/${consul::version}_web_ui/dist",
+        }
+      }
     }
+    'package': {
+      package { $consul::package_name:
+        ensure => $consul::package_ensure,
+      }
 
-    if $consul::ui_dir {
-      package { $consul::ui_package_name:
-        ensure => $consul::ui_package_ensure,
+      if $consul::ui_dir {
+        package { $consul::ui_package_name:
+          ensure => $consul::ui_package_ensure,
+        }
       }
     }
-
-  } else {
-    fail("The provided install method ${consul::install_method} is invalid")
+    'none': {}
```

Comment:
The readme is good for the full example, but can you update the inline docstring on init.pp:13?